### PR TITLE
core/vm: minor trivial clean up

### PR DESCRIPTION
**Pull request by lmittmann**

### CODE REVIEW

**Comment by MariusVanDerWijden** [comment]
I'm pretty sure the `IsPush` is an optimization. Do you have some numbers on the difference between your code and master?

**Review by holiman — APPROVED** [review]
LGTM

**Comment by lmittmann** [comment]
@MariusVanDerWijden  I could not find any significant performance difference.

```
name            old time/op  new time/op  delta
OpCodeIsPush-8  0.22ns ± 3%  0.22ns ± 5%  -0.22%  (p=0.025 n=47+46)
```

```go
func BenchmarkOpCodeIsPush(b *testing.B) {
	c := 0
	for i := 0; i < b.N; i++ {
		op := OpCode(i)
		if ok := op.IsPush(); ok {
			c++
		}
	}
}
```

**Comment by holiman** [comment]
Strange, I already wrote this once, but it seems to have disappeared. 

> I'm pretty sure the IsPush is an optimization.

Might or might not. Either way, it's pretty moot because it's not performance critical. It's only called from `instructionIterator.Next` in `asm.go` and `opObj.IsPush` in `goja.go`. It's also exposed as a library function, otherwise we might aswell remove it.

File: core/vm/opcodes.go [hunk 1]
```diff
@@ -25,11 +25,7 @@ type OpCode byte
 
 // IsPush specifies if an opcode is a PUSH opcode.
 func (op OpCode) IsPush() bool {
-	switch op {
-	case PUSH1, PUSH2, PUSH3, PUSH4, PUSH5, PUSH6, PUSH7, PUSH8, PUSH9, PUSH10, PUSH11, PUSH12, PUSH13, PUSH14, PUSH15, PUSH16, PUSH17, PUSH18, PUSH19, PUSH20, PUSH21, PUSH22, PUSH23, PUSH24, PUSH25, PUSH26, PUSH27, PUSH28, PUSH29, PUSH30, PUSH31, PUSH32:
-		return true
-	}
-	return false
+	return PUSH1 <= op && op <= PUSH32
```
Comment (chfast) [inline core/vm/opcodes.go:28]: `return int8(op) >= int8(PUSH1)`

**Pull request merged**